Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -213,6 +193,74 @@ public static string GetLanguageVersion(ModuleDefinition module, string targetFr
             }
         }
 
+        // Knowledge on unique machine Ids for different OSes
+        // obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = Environment.OSVersion.Platform;
+
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
```

Comment:
or should I just check for the only realistic platform that our users should be using (a.k.a. PlatformID.Win32S)